panel/io/state.py

```diff
@@ -209,6 +212,10 @@ class _state(param.Parameterized):
     _sessions = {}
     _session_key_funcs = {}
 
+    # Middlewares
+    _bokeh_event_middlewares: ClassVar[List[BokehEventMiddleware]] = []
```

Comment:
Should these be at the class-level or the instance level? Right now you define them at the class-level but require the user to register them on an instance.